### ISSUE TITLE
Original Demo should be Ported Completely to Kotlin

### ISSUE THREAD

**Issue by Mr00Anderson** [kotlin-graphics/imgui#90]
**Issue** 
I have found that not all things are working and some may be my error due. When copying some of the examples from the original libraries [demo](https://github.com/ocornut/imgui/blob/master/imgui_demo.cpp) they do not work. 

**Solution**
Port over the demo so that any changes in language and reference will not be left to the user to figure out, if the library is bugged or the example is not a direct translation. 

**Background**
I selected this because it was the only active library for intermediate that I saw. I even picked up Kotlin because of how much less work it takes. Lots of headbanging trying to figure out why examples in the demo wont work or error out. I could try to help contribute, though still learning Kotlin.

**Comment by Mr00Anderson** [comment]
Somehow I found them. My bad wow.

**Comment by elect86** [comment]
Hey Daniel,

shall we modify/add some lines to the readme to make it clear?

**Comment by Mr00Anderson** [comment]
I did not have the sources loaded correctly in intelliJ so when I "Ctrl+B" and was unable to browse to the demo.  Cloned the demo code and was able to get my project working. 

I figured out why my UI was not working. I was confusion between what is now under the package named DSL in the recent commits and the ImGui methods. It was a result of directly translating the Java example to kotlin and using the c++ API and not realizing that the DSL was calling multiple functions vs the ImGui c++ API. I was thinking about how to explain it in the readme that there is some wrapped functionality that if you use the ImGui C+ API will not match the same method calls in this library.
